src/scheduler/MpiWorld.cpp

```diff
@@ -176,6 +176,54 @@ std::string MpiWorld::getHostForRank(int rank)
     return rankHostMap[rank];
 }
 
+void MpiWorld::getCartesianRank(int rank, int* dims, int* periods, int* coords)
+{
+    if (rank > this->size - 1) {
+        throw std::runtime_error(
+          fmt::format("Rank {} bigger than world size {}", rank, this->size));
+    }
+
+    // Compute the corresponding rank in a 2-dim grid given the original process
+    // rank.
+    // Note - this operation, when restricted to 2dims is quivalent to:
+    // coords = {rank / sideLength, rank % sideLength}
+    int sideLength = static_cast<int>(std::floor(std::sqrt(this->size)));
+    int nprocs = sideLength * sideLength;
+    if (rank >= nprocs) {
+        for (uint8_t i = 0; i < MPI_CART_MAX_DIMENSIONS; i++) {
+            dims[i] = sideLength;
+            periods[i] = 0;
+            coords[i] = MPI_UNDEFINED;
+        }
+    } else {
+        for (uint8_t i = 0; i < MPI_CART_MAX_DIMENSIONS; i++) {
+            nprocs /= sideLength;
+            dims[i] = sideLength;
+            periods[i] = 0;
+            coords[i] = rank / nprocs;
+            rank %= nprocs;
+        }
+    }
+}
+
+void MpiWorld::getRankFromCoords(int* rank, int* coords)
+{
+    int sideLength = static_cast<int>(std::floor(std::sqrt(this->size)));
+    int prank = 0;
+    int factor = 1;
+
+    int i = MPI_CART_MAX_DIMENSIONS - 1;
+    for (; i >= 0; --i) {
```

Comment:
Then the style of this for loop is also different, we usually initialise loop variables in the loop declaration (and this is using an int whereas the one above is using `uint8_t`)

Reply:
Funny enough, I had to change the type in this case (as using an unsigned int would overflow) but didn't bother looking what was used elsewhere.